controllers/keda/scaledobject_controller_test.go

```diff
@@ -1028,6 +1028,137 @@ var _ = Describe("ScaledObjectController", func() {
 			return k8sClient.Get(context.Background(), types.NamespacedName{Name: fmt.Sprintf("keda-hpa-%s", soName), Namespace: "default"}, hpa)
 		}).Should(HaveOccurred())
 	})
+
+	// Fix issue 5281
+	It("reconciles scaledobject when hpa spec is changed", func() {
```

Comment:
Could you please also add a case for labels and annotations? So we are fully covered here.

Reply:
Added.